### PR TITLE
Fix resolution of lerna packages in buildutils

### PR DESCRIPTION
**Pull request by vidartf**
## References

Fixes an error in #6720 .

## Code changes

The previous code was a but rushed for 1.0. This should resolve the issue properly.

## User-facing changes

None. It will be easier for extension devs with lerna repos to update lab.

## Backwards-incompatible changes

None.

### CODE REVIEW

File: buildutils/src/utils.ts [hunk 1]
```diff
@@ -15,10 +15,13 @@ export function getLernaPaths(basePath = '.'): string[] {
...
   let paths: string[] = [];
-  let packages =
-    baseConfig.workspaces.packages ||
-    baseConfig.workspaces ||
-    baseConfig.packages;
+  let packages;
+  if (baseConfig.workspaces) {
+    packages = baseConfig.workspaces.packages || baseConfig.workspaces;
+  } else {
+    let baseConfig = require(path.join(basePath, 'lerna.json'));
```
Comment (blink1073) [inline buildutils/src/utils.ts]: Is this line redundant?

Comment (vidartf) [reply buildutils/src/utils.ts]: `lerna.json` is the significant part here (that actually fixes the issue).

Comment (vidartf) [reply buildutils/src/utils.ts]: Possibly we would want a try/except to catch the file not existing, before giving a helpful error message.

**Comment by vidartf** [comment]
Note: Currently the same test is timing out in both Win and Linux JS test:

```
@jupyterlab/notebook
NotebookActions
× "before all" hook in "NotebookActions"
HeadlessChrome 76.0.3803
```

I don't think this should be related to this PR though?

**Comment by jasongrout** [comment]
> I don't think this should be related to this PR though?

That's similar to an error that has been timing out on the *windows* js tests for a while.

**Comment by blink1073** [comment]
I tried kicking it.

**Comment by blink1073** [comment]
Thanks!

**Pull request merged**